plugins/processors/starlark/testdata/sparkplug.star

```diff
@@ -0,0 +1,305 @@
+
+# [[inputs.mqtt_consumer]]
+#   servers = ["tcp://your_server:1883"]
+#   qos = 0
+#   connection_timeout = "30s"
+#   topics = ["spBv1.0/#"]
+#   persistent_session = false
+#   client_id = ""
+#   username = "your username"
+#   password = "your password"
+# 
+#   # Sparkplug protobuf configuration
+#   data_format = "xpath_protobuf"
+#    
+#   # URL of sparkplug protobuf prototype
+#   xpath_protobuf_type = "org.eclipse.tahu.protobuf.Payload"
+#    
+#   # Location of sparkplug_b.proto file
+#   xpath_protobuf_file = "/apps/telegraf/config/sparkplug_b.proto"
+# 
+#   [[inputs.mqtt_consumer.xpath_protobuf]]
+#     metric_selection = "metrics[not(template_value)]"
+#     metric_name = "concat('edge', substring-after(name, ' '))"
+#     timestamp = "timestamp"
+#     timestamp_format = "unix_ms"
+#     [inputs.mqtt_consumer.xpath_protobuf.tags]
+#       name = "substring-after(name, ' ')"
+#     [inputs.mqtt_consumer.xpath_protobuf.fields_int]
+#       type = "datatype"
+#       alias = "alias"
+#     [inputs.mqtt_consumer.xpath_protobuf.fields]
+#       # A metric value must be numeric
+#       value = "number((int_value | long_value | float_value | double_value | boolean_value))"
+#       name = "name"
+# 
+# # Starlark processor
+# [[processors.starlark]]
+#   script = "sparkplug.star"
+# 
+#   # Optionally Define constants used in sparkplug.star
+#   # Constants can be defined here or they can be defined in the 
+#   # sparkplug_b.star file.
+#
+#   [processors.starlark.constants]     
+#     # Tags used to identify message type - 3rd field of topic
+#     BIRTH_TAG = "BIRTH/"
+#     DATA_TAG = "DATA/"
+# 
+#     # Number of messages to hold if alias cannot be resolved 
+#     MAX_UNRESOLVED = 3
+# 
+#     # Provide alternate names for the 5 sparkplug topic fields.  
+#     # The topic contains 5 fields separated by the '/' character.  
+#     # Define the tag name for each of these fields.
+#     MSG_FORMAT = "false"        #0
+#     GROUP_ID   = "reactor"      #1
+#     MSG_TYPE   = "false"        #2
+#     EDGE_ID    = "edgeid"       #3
+#     DEVICE_ID  = "experiment"   #4
+#
+
+BIRTH_TAG = "BIRTH/"
+DEATH_TAG = "DEATH/"
+DATA_TAG  = "DATA/"
+  
+# Number of messages to hold if alias cannot be resolved 
+MAX_UNRESOLVED = 3
+
+# Provide alternate names for the 5 sparkplug topic fields.  
+# The topic contains 5 fields separated by the '/' character.  
+# Define the tag name for each of these fields.
+MSG_FORMAT = "false"        #0
+GROUP_ID   = "Reactor"      #1
+MSG_TYPE   = "MsgType"      #2
+EDGE_ID    = "EdgeID"       #3
+DEVICE_ID  = "Experiment"   #4
+ 
+########### Begin sparkplug.star script
+
+
+load("logging.star", "log")
+
+state = {
+    "aliases":    dict(),
+    "devices":    dict(),
+    "unresolved": list()
+}
+
+def extractTopicTags(metric):
+    msg_format   = 'xx'
+    groupid      = 'xx'
+    msg_type     = 'xx'
+    edgeid       = 'xx'
+    deviceid     = 'xx'
```

Comment:
Is this a placeholder? If so, please use some more illustrative values.

Reply:
I changed them to a null string.